### ISSUE TITLE
Premission issues in MQ IPT Docker image

### ISSUE THREAD

**Issue by konradloch** [ibm-messaging/mq-container#580]
After following image build instruction for MQ IPT (https://github.com/ibm-messaging/mq-container/tree/master/incubating/mqipt) you can encounter in running container two permission issues:

1. IOException reading //AMQJAVA.LOG
```
IOException reading //AMQJAVA.LOG
java.io.IOException: Permission denied 
```
That log appear twice at container start then it is not visible. Prefix '//' suggest that some path is not set?

2. Couldn't create user preferences directory. User preferences are unusable. 
```
WARNING: Couldn't create user preferences directory. User preferences are unusable.  
Sep 16, 2024 2:09:22 PM java.util.prefs.FileSystemPreferences$1 run
WARNING: java.io.IOException: No such file or directory
Sep 16, 2024 2:09:53 PM java.util.prefs.FileSystemPreferences checkLockFile0ErrorCode                                                                              
WARNING: Could not lock User prefs.  Unix error code 2.
Sep 16, 2024 2:09:53 PM java.util.prefs.FileSystemPreferences syncWorld                                                                                             
WARNING: Couldn't flush user prefs: java.util.prefs.BackingStoreException: Couldn't get file lock. 
```

Those logs appear every 30s and might be distractious. Quick fix for that is create home directory for 1001 USER (used in dockerfile) but I am not sure if this is the correct solution. 

(used MQ IPT version: 9.2.0.27 LinuxX64, I don't know if it occurs in other versions)

I don't see any problem with MQ IPT itself. The question is whether these two errors do not cause some sort of bug.

**Comment by KNishit92** [comment]
Hi,
You can make the change in your own fork.

**Comment by KNishit92** [comment]
You seem to be using MQIPT 9.2.0.27.
The first issue  has already been fixed in MQIPT 9.3.2. MQIPT doesn't need the AMQJAVA.LOG file, and from MQPIT 9.3.2 onwards it doesn't get created. We suggest you should upgrade to MQIPT 9.4.
The second issue is because the JRE tries to create a user preferences directory, but there is no home directory available. It's not needed by MQIPT, rather, it's something that the JRE does automatically. It seems that your suggested fix is correct – if the user home directory is present and writeable the error disappears. The suggested change could be made in your own fork, or,
you can alternatively add the following options system properties:
-Djava.util.prefs.userRoot=/tmp
-Djava.util.prefs.systemRoot=/tmp
